llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -296,11 +297,35 @@ void saveModuleIR(Module &M, StringRef OutFilename) {
   MPM.run(M, MAM);
 }
 
-std::string saveModuleIR(Module &M, int I, StringRef Suff) {
-  DUMP_ENTRY_POINTS(M, EmitOnlyKernelsAsEntryPoints, "saving IR");
+std::unique_ptr<Module> makeDummyImageIR(const Module &M) {
```

Comment:
When we do code split we don't clone the whole module, but we only take what's necessary.

As I understand it, the input here is a device image which contains virtual function definitions. But the code below suggests that it also may contain other functions which are called by virtual functions. If we can save on copying them, then we probably should.